### DISCUSSION TITLE
Typescript type issue.

### DISCUSSION THREAD

**Discussion by Horsetopus** [prisma/prisma#9908] [Q&A]
Hi,

I'm using Prisma and I like the whole thing, but I have some issues with matching a Typescript definition with the one generated by Prisma.

Please lend me a hand.

Here is my shema.prisma

> enum FieldType {
>   number
>   text
>   boolean
> }
> 
> model ResourceDefinition {
>   id          Int               @id @default(autoincrement())
>   name        String            @unique
>   description String
>   fields      FieldDefinition[]
> }
> 
> model FieldDefinition {
>   id                   Int                @id @default(autoincrement())
>   resourceDefinitionId Int
>   resourceDefinition   ResourceDefinition @relation(fields: [resourceDefinitionId], references: [id])
>   name                 String
>   description          String
>   type                 FieldType
>   parameters           Json
> 
>   @@unique([resourceDefinitionId, name])
> }
> 

I use a MySQL provider

Here is my Typescript code:

> type Field = {
> 
>     id:number;
>     name:string;
>     description:string;
>     type:string;
>     parameters:Prisma.JsonArray
> }
> 
> type Resource = {
> 
>     id:number;
>     name:string;
>     description:string;
>     fields?:Field[];
> }
> 
> const registryService = {
> 
>     async addResourceDefinition( name:string, description:string, fields:Field[] ):Promise<number> {
> 
>         const namespace = await prisma.resourceDefinition.create( { data: { name, description, fields } } );
> 
>         return namespace.id;
>     }
> };

The problem is with the fields parameter in addResourceDefinition.
const namespace = await prisma.resourceDefinition.create( { data: { name, description, **fields** } } );

I get the error:
`<html>TS2322: Type 'Field[]' is not assignable to type 'FieldDefinitionUncheckedCreateNestedManyWithoutResourceDefinitionInput | FieldDefinitionCreateNestedManyWithoutResourceDefinitionInput'.<br/>Type 'Field[]' has no properties in common with type 'FieldDefinitionCreateNestedManyWithoutResourceDefinitionInput'.`

But I can't figure it out.
Any help is appreciated.
Thank you.

**Comment by austincrim** [comment]
Hey @Horsetopus!

The type error is occurring because you are actually doing a [nested `createMany`](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries/#create-a-single-record-and-multiple-related-records) operation on the `fields` attribute. Here's an updated snippet that worked for me:
```typescript
import { Prisma } from '@prisma/client'

async function addResourceDefinition(
  name: string,
  description: string,
  fields: Prisma.FieldDefinitionCreateManyInput[]
): Promise<number> {
  const namespace = await prisma.resourceDefinition.create({
    data: { name, description, fields: { createMany: { data: fields } } }
  })

  return namespace.id
}
```
Also please note that you can use Prisma's generate type definitions instead of creating your own, if you prefer.
```typescript
import type { FieldDefinition, ResourceDefinition } from '@prisma/client
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
